libraries/AP_Mount/AP_Mount_Tusuav.cpp

```diff
@@ -0,0 +1,1124 @@
+                        (unsigned char)_parsed_msg.msg.data[5], 
+                        (unsigned char)_parsed_msg.msg.data[6],
+                        (unsigned char)_parsed_msg.msg.data[7]); 
+                            
+                GCS_SEND_TEXT(MAV_SEVERITY_INFO, "%s BootVersion: %d.%d.%d %02d-%02d-%02d %02d:%02d", send_text_prefix,                                    
+                        (unsigned char)_parsed_msg.msg.data[8],
+                        (unsigned char)_parsed_msg.msg.data[9],
+                        (unsigned char)_parsed_msg.msg.data[10],
+                        (unsigned char)_parsed_msg.msg.data[11],
+                        (unsigned char)_parsed_msg.msg.data[12],
+                        (unsigned char)_parsed_msg.msg.data[13],
+                        (unsigned char)_parsed_msg.msg.data[14], 
+                        (unsigned char)_parsed_msg.msg.data[15]);
+                    break;
+            }
+            break;
+        }
+        default:
+            break;
+        }
+        break;
+    }
+
+    case GimbalCmd::MasCmd_Load:{
+
+        break;
+    }
+
+    default:
+        break;
+    }
+}
+
+// send packet to gimbal.  total_len includes everything including the header '0x5A' and crc
+// returns true on success, false if outgoing serial buffer is full or other errors
+bool AP_Mount_Tusuav::send_packet(const struct FrameType &frame)
+{
+    if (!_initialised) {
+        return false;
+    }
+
+    if (frame.data == NULL && frame.data_len > 0) {
+        return false;
+    }
+
+    AP_Mount_Tusuav::FrameType modifiable_frame = frame;
+    modifiable_frame.total_len = modifiable_frame.data_len + AP_MOUNT_TUSUAV_PACKETLEN_MIN;
+    // calculate and sanity check packet size
+    if (frame.total_len >= AP_MOUNT_TUSUAV_PACKETLEN_MAX) {
+        debug("send_packet data buff too large");
+        return false;
+    }
+
+    // check for sufficient space in outgoing buffer
+    if (_uart->txspace() < frame.total_len) {
+        debug("tx space too low (%u < %u)", (unsigned)_uart->txspace(), (unsigned)frame.total_len);
+        return false;
+    }
+
+    // buffer for holding outgoing packet
+    uint8_t send_buff[frame.total_len];
+    uint8_t send_buff_ofs = 0;
+
+    // packet header length cmd and sub-cmd
+    send_buff[send_buff_ofs++] = AP_MOUNT_TUSUAV_HEADER1;
+    send_buff[send_buff_ofs++] = frame.total_len;
+    send_buff[send_buff_ofs++] = frame.flag.all;
+    send_buff[send_buff_ofs++] = frame.cmd;
+    send_buff[send_buff_ofs++] = frame.sub_cmd;
+
+    // data
+    if (frame.data_len > 0) {
+        auto* data_ptr = reinterpret_cast<const uint8_t*>(frame.data);
+        for (size_t i = 0; i < frame.data_len; ++i) {
+            send_buff[send_buff_ofs + i] = data_ptr[i];
+        }
+        send_buff_ofs += frame.data_len;
+    }
+
+    // crc
+    send_buff[send_buff_ofs] = crc_calculate(send_buff, send_buff_ofs);
+    send_buff_ofs++;
```

Comment:
Is this actually working?!

You're assigning a 16-bit checksum into an 8-bit field here....